src/compile.rs

```diff
@@ -524,9 +524,22 @@ pub(crate) fn compile_inner(inner_re: &str, options: &RegexOptions) -> Result<Ra
 }
 
 /// Compile the analyzed expressions into a program.
-pub fn compile(info: &Info<'_>) -> Result<Prog> {
+pub fn compile(info: &Info<'_>, anchored: bool) -> Result<Prog> {
     let mut c = Compiler::new(info.end_group);
+    if !anchored {
+        c.b.add(Insn::Split(3, 1));
```

Comment:
Maybe add some comments about these magic numbers, i.e. 3 means after these added instructions, 1 is the Any.

Reply:
good idea, it is much clearer what is happening that way - done